src/Tests/Microsoft.Diagnostics.Monitoring.UnitTests/AuthenticationTests.cs

```diff
@@ -0,0 +1,232 @@
+
+            var metrics = await client.GetMetricsAsync(DefaultApiTimeout);
+            Assert.NotNull(metrics);
+        }
+
+        /// <summary>
+        /// Tests the behavior of the mertics URL address, namely that all routes will return
+        /// 404 Not Found except for the metrics route (200 OK).
+        /// </summary>
+        [Fact]
+        public async Task MetricsAddressTest()
+        {
+            await using DotNetMonitorRunner toolRunner = new DotNetMonitorRunner(_outputHelper);
+
+            await toolRunner.StartAsync(DefaultStartTimeout);
+
+            using ApiClient client = new ApiClient(_outputHelper, await toolRunner.GetMetricsAddressAsync(DefaultAddressTimeout));
+
+            // Any non-metrics route on the metrics address should 404 Not Found
+            var statusCodeException = await Assert.ThrowsAsync<ApiStatusCodeException>(
+                () => client.GetProcessesAsync(DefaultApiTimeout));
+            Assert.Equal(HttpStatusCode.NotFound, statusCodeException.StatusCode);
+
+            // TODO: Verify other routes (e.g. /dump, /trace, /logs) also 404 Not Found
+
+            // Metrics should not throw on the metrics address
+            var metrics = await client.GetMetricsAsync(DefaultApiTimeout);
+            Assert.NotNull(metrics);
+        }
+
+        /// <summary>
+        /// Tests that turning off authentication allows access to all routes without authentication.
+        /// </summary>
+        [Fact]
+        public async Task DisableAuthenticationTest()
+        {
+            await using DotNetMonitorRunner toolRunner = new DotNetMonitorRunner(_outputHelper);
+            toolRunner.DisableAuthentication = true;
+            await toolRunner.StartAsync(DefaultStartTimeout);
+
+            using ApiClient client = new ApiClient(_outputHelper, await toolRunner.GetDefaultAddressAsync(DefaultAddressTimeout));
+
+            // Check that /processes does not challenge for authentication
+            var processes = await client.GetProcessesAsync(DefaultApiTimeout);
+            Assert.NotNull(processes);
+
+            // Check that /metrics does not challenge for authentication
+            var metrics = await client.GetMetricsAsync(DefaultApiTimeout);
+            Assert.NotNull(metrics);
+        }
+
+        /// <summary>
+        /// Tests that API key authentication can be configured correctly and
+        /// that the key can be rotated wihtout shutting down dotnet-monitor.
+        /// </summary>
+        [Fact]
+        public async Task ApiKeyAuthenticationSchemeTest()
+        {
+            const string AlgorithmName = "SHA256";
+
+            await using DotNetMonitorRunner toolRunner = new DotNetMonitorRunner(_outputHelper);
+
+            _outputHelper.WriteLine("Generating API key.");
+
+            // Generate initial API key
+            byte[] apiKey = GenerateApiKey();
+
+            // Set API key via key-per-file
+            RootOptions options = new();
```

Comment:
Now to start the debate between:
var options = new RootOptions() vs. RootOptions options = new()?

Reply:
What do you think? I can switch all of the test code to be either way.

Reply:
I have no strong feelings either way.

Reply:
I've changed all of the test code to remove the type from the ctor call if the type is explicitly used on the member of local.